guides/v2.1/coding-standards/docblock-standard-general.md

```diff
@@ -492,21 +497,24 @@ public function deleteDirectory($path)
 #### @return tag
 {:#return}
 
-If there is no explicit return statement in a method or function, a `@return void` should be used in the documentation.
+In general method return type signatures should be prefered over `@return` type annotations.
+If that is not possible due to ambiguous return types, the `@return` type annotation must be used.
+
+If there is no explicit return statement in a method or function or a return statement without a value, a `void` return type must be declared in the method signature.
 
-If the method returns itself, `return $this` should be used.
+If the method returns itself, `@return $this` must be used, unless the return type is specified in the method signature.
```

Comment:
So, what to use for `function FooBar(): self` ? I think no `@return` needed in this case.

When there is not type hint, I'm a bit concerned whether we need explicit `@return $this` or PhpStorm may resolve correct type anyway.

Reply:
Good point. I'll expand the section accordingly.